group-service/src/main/kotlin/zhi/yest/vk/friendfinder/security/jwt/VkJwtDecoder.kt

```diff
@@ -0,0 +1,14 @@
+package zhi.yest.vk.friendfinder.security.jwt
+
+import org.springframework.security.oauth2.jwt.Jwt
+import org.springframework.security.oauth2.jwt.ReactiveJwtDecoder
+import reactor.core.publisher.Mono
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+class VkJwtDecoder : ReactiveJwtDecoder {
+    override fun decode(token: String?): Mono<Jwt> {
+        // TODO: complete this once gateway sends full details
+        return Mono.just(Jwt(token, Instant.now(), Instant.now().plus(10, ChronoUnit.DAYS), mapOf("some" to "header"), mapOf("sub" to "To Do")))
```

Comment:
what deteails will we keep in JWT token?

Reply:
In my opinion, it would make sense to also add name, surname and user id, as well as issuing time and expiration time.